### PR TITLE
IB/RC/CM: Optimize address size.

### PR DESCRIPTION
**Pull request by yosefe**

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/1092/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1695/ for details.

**Comment by yosefe** [comment]
@alinask @alex-mikheev can you pls review?

File: src/uct/ib/rc/base/rc_ep.h [hunk 1]
```diff
@@ -57,6 +57,11 @@ struct uct_rc_ep {
 UCS_CLASS_DECLARE(uct_rc_ep_t, uct_rc_iface_t*);
 
 
+typedef struct uct_rc_ep_address {
+    uint8_t             qp_num[3];
+} UCS_S_PACKED ucp_rc_ep_address_t;
```
Comment (alinask) [inline src/uct/ib/rc/base/rc_ep.h]: can you please explain why this is 3?

Comment (yosefe) [inline src/uct/ib/rc/base/rc_ep.h]: IB QP number is 24 bits, so we save one byte by using array of 3 bytes instead of uint32_t

**Comment by alinask** [comment]
:+1:

**Pull request merged**